### PR TITLE
End time after start time

### PR DESCRIPTION
**Pull request by joycelynng**

### CODE REVIEW

**Comment by joycelynng** [comment]
Issue #185

File: src/main/java/seedu/address/logic/LogicManager.java [hunk 1]
```diff
@@ -29,9 +29,13 @@ public LogicManager(Model model, Storage storage) {
...
-        Command command = parser.parseCommand(commandText);
-        command.setData(model);
-        return command.execute();
+        try{
+        	Command command = parser.parseCommand(commandText);
+            command.setData(model);
+            return command.execute();
+        } catch (Exception ex){
+        	return new CommandResult(ex.getMessage());
+        }
```
Comment (kabir-plod) [inline src/main/java/seedu/address/logic/LogicManager.java]: Don't think this is necessary. If something goes wrong in the execute method, the exception should be handled there. The exception message can be supplied to the CommandResult returned by the command class

File: src/main/java/seedu/address/logic/commands/EditCommand.java [hunk 2]
```diff
@@ -89,27 +82,34 @@ public CommandResult execute() {
...
-            	}
+            	Task.validateEndDateTimeAfterStartDateTime(startDateTime, endDateTime);
             }
             
-            if(!newStartDateTime.isPresent() && newEndDateTime.isPresent()){
-            	LocalDateTime endDateTime = newEndDateTime.get();
+            if(newStartDateTime.isPresent() && !newEndDateTime.isPresent()){
+            	if(isRemoveEndDateTime){
+            		throw new UnsupportedOperationException(Task.MESSAGE_START_DATE_TIME_CANNOT_BE_SET_WITH_END_DATE_TIME_REMOVED);
+            	}
```
Comment (kabir-plod) [inline src/main/java/seedu/address/logic/commands/EditCommand.java]: Return CommandResult with the same message instead. Alternatively you could move the validation to the command constructor. If it fails, an exception is thrown for the parser to catch - an IncorrectCommand is returned straight away.

File: src/test/java/seedu/address/logic/parser/ParserTest.java [hunk 3]
```diff
@@ -113,7 +113,7 @@ public void parseCommand_addEventNoDate_addCommandReturned() {
 	
 	@Test
 	public void parseCommand_addEventValidOrder_addCommandReturned() {
-		String userInput = "add 'party' from 5:00 to 5:00 on 12-12-12";
+		String userInput = "add 'party' from 5:00 to 10:00 on 12-12-12";
 		Command command = parser.parseCommand(userInput);
```
Comment (kabir-plod) [inline src/test/java/seedu/address/logic/parser/ParserTest.java]: Add another test to check that that an incorrect command is indeed returned

**Pull request merged**